surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/SmartTestingSurefireProvider.java

```diff
@@ -64,14 +64,17 @@ private TestsToRun getTestsToRun(Object forkTestSet) throws TestSetFailedExcepti
     private TestsToRun getOptimizedTestsToRun(TestsToRun testsToRun) {
         Set<TestSelection> selection = SmartTesting
             .with(className -> testsToRun.getClassByName(className) != null)
-            .in(getBasedir())
+            .in(getProjectDir())
             .applyOnClasses(testsToRun);
 
         return new TestsToRun(SmartTesting.getClasses(selection));
     }
 
-    private String getBasedir() {
-        final String path = this.bootParams.getReporterConfiguration().getReportsDirectory().getPath();
-        return path.substring(0, path.indexOf(File.separator + "target"));
+    private File getProjectDir() {
+        if (System.getProperty("basedir") == null) {
+            return bootParams.getTestRequest().getTestSourceDirectory().getParentFile().getParentFile().getParentFile();
```

Comment:
Humm... is it always only three levels up? Maybe recursively until we find first `pom.xml`? I know it's unlikely to have it more nested, but why not making it ready for that?

Reply:
The reason why I did it in this way is the case when you don't use the default name for the pom file. But yeah, this is really rare and mainly, usually, you have the default `pom.xml` file there and another (custom) one next to it...

Reply:
Fixed  anyway